### PR TITLE
Fix compiler warnings and lsquit() error

### PR DESCRIPTION
**Pull request by bingao**
One unused variable is removed. C++ preprocessor is added so that lsquit() will be called only when `BUILT_IN_LSDALTON` is `ON`.

### CODE REVIEW

**Comment by bast** [comment]
Thanks! I will review and if possible merge in a moment but please also review and if possible merge mine: https://github.com/openrsp/openrsp/pulls :-)

File: TODO.rst [hunk 1]
```diff
@@ -6,6 +6,13 @@ TODO List
 Short-term
 ----------
 
+To be done before release:
```
Comment (bast) [inline TODO.rst:9]: These should be issues, not tracked in the repo IMO.

**Comment by bast** [comment]
Thanks for the fix - can you please also remove the spurious merge commit?

**Comment by bingao** [comment]
> Thanks for the fix - can you please also remove the spurious merge commit?

What is the "spurious merge commit"?

**Comment by bast** [comment]
Sorry, I meant the one merge commit up there. There is no reason why we need to carry it around in the history IMO. In other words I suggest you rebase your change on top of `master`. Of course I can also do this when merging later through the UI.

File: TODO.rst (hunk 1 above)
Comment (bast) [reply TODO.rst:9]: But if this slows down this PR, you can also create an issue to convert `TODO.rst` into issues :-)

**Comment by bingao** [comment]
OK. Actually, I did not know how to do it on top of `master`. Now I work on a fork of `master` and create pull request once I finish my work.

If you have time, would mind tell me how to do it? Thank you.

**Comment by bast** [comment]
You can rebase it like this:
```
git checkout master
git pull git@github.com:openrsp/openrsp.git master

git checkout BinGao/fix_compiler_warnings
git rebase master

# assuming origin is your fork, also we need to force-push since history changed
git push -f origin BinGao/fix_compiler_warnings
```

File: TODO.rst (hunk 1 above)
Comment (bast) [reply TODO.rst:9]: Followed up in #14.

**Review by bast — APPROVED** [review]
I will rebase and squash through the UI since I need to move on with my upcoming PR :-)

**Comment by bingao** [comment]
Hei @bast 

Thank you. Can I work on other issues regarding OpenRSP and do not need to wait for the rebase?

**Comment by bast** [comment]
It's already done.

**Comment by bingao** [comment]
Great. Thank you. Then I will delete this branch, is it Ok?

**Comment by bast** [comment]
Yes, the PR is merged, this means you can remove the source branch.

**Pull request merged**